versioned/persist/nontx/src/main/java/org/projectnessie/versioned/persist/nontx/NonTransactionalDatabaseAdapter.java

```diff
@@ -853,4 +997,72 @@ protected abstract GlobalStateLogEntry fetchFromGlobalLog(
 
   protected abstract List<GlobalStateLogEntry> fetchPageFromGlobalLog(
       NonTransactionalOperationContext ctx, List<Hash> hashes);
+
+  protected Hash writeRefLogEntry(
+      NonTransactionalOperationContext ctx,
+      String refName,
+      RefLogEntry.RefType refType,
+      Hash parentRefLogId,
+      Hash commitHash,
+      RefLogEntry.Operation operation,
+      long timeInMicros,
+      ByteString details)
+      throws ReferenceConflictException {
+
+    Hash currentRefLogId = randomHash();
+    org.projectnessie.versioned.RefLogEntry parentEntry = fetchFromRefLog(ctx, parentRefLogId);
+    RefLogEntry refLogEntry =
+        buildRefLogEntry(
+            refName,
+            refType,
+            parentRefLogId,
+            commitHash,
+            operation,
+            timeInMicros,
+            details,
+            currentRefLogId,
+            parentEntry);
+
+    writeRefLog(ctx, refLogEntry);
+
+    return currentRefLogId;
+  }
+
+  private RefLogEntry buildRefLogEntry(
```

Comment:
This should live in `ABstractDatabaseAdapter` like `buildIndividualCommit` does

Reply:
I intentionally didn't added there to avoid having `ReflogEntry` to `adaptertypes.ReflogEntry` conversion. If I keep it inside like `writeGlobalCommit`, I can avoid that extra conversion as I can build `adaptertypes.ReflogEntry `directly.

So, I am basically following like `writeGlobalCommit`